oneflow/core/framework/nn_graph.cpp

```diff
@@ -239,6 +240,20 @@ Maybe<void> NNGraph::RegisterNewVariableOpInJobPass() {
   }));
   return Maybe<void>::Ok();
 }
+Maybe<void> NNGraph::DeleteOutdatedVariableInVariableTensorMgr() {
+  std::set<std::string> variable_names;
+  OpGraph op_graph(job_);
+  JUST(op_graph.MaybeForEachNode([&](OpNode* op_node) -> Maybe<void> {
+    if (op_node->op().op_conf().has_variable_conf() == false) { return Maybe<void>::Ok(); }
+    variable_names.insert(op_node->op().op_name());
+    return Maybe<void>::Ok();
+  }));
```

Comment:
[Use lambdas for complex initialization, especially of const variables](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es28-use-lambdas-for-complex-initialization-especially-of-const-variables) ，这样做的优势在链接里有详细说明

```suggestion
  const std::set<std::string> variable_names = [&]() {
    std::set<std::string> val;
    OpGraph op_graph(job_);
    JUST(op_graph.MaybeForEachNode([&](OpNode* op_node) -> Maybe<void> {
      if (op_node->op().op_conf().has_variable_conf() == false) { return Maybe<void>::Ok(); }
      val.insert(op_node->op().op_name());
      return Maybe<void>::Ok();
    }));
    return val;
  }();
```

Reply:
好的